### ISSUE TITLE
Water polygons are not merged

### ISSUE THREAD

**Issue by westnordost** [pnorman/tilekiln#38]
The StreetComplete style draws lines around water bodies as "shore lines". The various docks and canals in Hamburg are mapped in OSM each as individual neighboring polygons.

### Current behavior

Neighbouring water polygons are not merged. The area around [Ellerholzbrücke](https://www.openstreetmap.org/#map=18/53.52589/9.97342) looks like this:

![ellerholzbruecke-tilekiln](https://github.com/pnorman/tilekiln/assets/4661658/29a7eff8-bd16-41b9-96fe-6add3b77eeb7)

See https://streetcomplete.app/map-osm/

### Expected behavior

Neighbouring water polygons are merged. The area around [Ellerholzbrücke](https://www.openstreetmap.org/#map=18/53.52589/9.97342) looks like this with the JawgMaps tiles:

![ellerholzbruecke-jawg](https://github.com/pnorman/tilekiln/assets/4661658/4db0bc1f-fc1b-492f-b45f-a4ffbeb92c42)

See https://streetcomplete.app/map-jawg/

**Comment by pnorman** [comment]
I should merge them for tile size, but it's valid to have them unmerged. In fact, if you're dealing with multiple water=* values, they *can't* be fully merged.

**Comment by westnordost** [comment]
True, I think jawg just throws all water into one bucket. Shortbread doesn't.

**Comment by westnordost** [comment]
Not sure what you mean.

**Comment by pnorman** [comment]
What you're getting will always end up happening when styling the outer line of a polygon.

**Comment by westnordost** [comment]
But is there another way?

**Comment by pnorman** [comment]
Yes - a larger fill followed by a smaller inner, the same way we do casings on roads.

**Comment by westnordost** [comment]
Oh, I have been using "line-gap-width" for the casing. But anyway, when using two fill layers with one of them being slightly offset to be smaller, it sounds like it would result in the same rendering as when using a offset line for the "shore".
